### PR TITLE
Add min_samples and n_top_classes to TrainTestPerformance in NLP

### PR DESCRIPTION
**Pull request by nirhutnik**
image

### CODE REVIEW

File: deepchecks/nlp/checks/model_evaluation/train_test_performance.py [hunk 1]
```diff
@@ -33,6 +33,17 @@ class TrainTestPerformance(TrainTestPerformanceAbstract, TrainTestCheck):
     scorers: Union[Mapping[str, Union[str, Callable]], List[str]], default: None
         Scorers to override the default scorers, find more about the supported formats at
         https://docs.deepchecks.com/stable/user-guide/general/metrics_guide.html
+    min_samples: int, default: 30
+        Minimum number of samples required to calculate model performance. If scorer is per class, then
+        this is the minimum number of samples per class.
+    n_top_classes: int, default: 10
+        Number of top classes to show in the label graphs. The top classes are determined by the show_classes_by
+        parameter. if None, then all classes are shown.
+    show_classes_by: str, default: 'test_largest'
```
Comment (Nadav-Barak) [inline deepchecks/nlp/checks/model_evaluation/train_test_performance.py:42]: Maybe also option of largest diff?

File: deepchecks/nlp/checks/model_evaluation/train_test_performance.py [hunk 2]
```diff
@@ -33,6 +33,17 @@ class TrainTestPerformance(TrainTestPerformanceAbstract, TrainTestCheck):
     scorers: Union[Mapping[str, Union[str, Callable]], List[str]], default: None
         Scorers to override the default scorers, find more about the supported formats at
         https://docs.deepchecks.com/stable/user-guide/general/metrics_guide.html
+    min_samples: int, default: 30
```
Comment (Nadav-Barak) [inline deepchecks/nlp/checks/model_evaluation/train_test_performance.py:36]: hmm, maybe 10? for score per class I think its a bit harsh (sometime we will be evaluated on a small dataset)

File: deepchecks/utils/abstracts/train_test_performace.py [hunk 3]
```diff
@@ -137,3 +163,35 @@ def add_condition_class_performance_imbalance_ratio_less_than(
         name = f"Relative ratio difference between labels '{score}' score is less than {format_percent(threshold)}"
         condition = get_condition_class_performance_imbalance_ratio_less_than(threshold=threshold, score=score)
         return self.add_condition(name=name, condition_func=condition)
+
+# def _filter_classes_without_enough_samples(results_df, class_list, train_dataset_name, test_dataset_name) \
```
Comment (Nadav-Barak) [inline deepchecks/utils/abstracts/train_test_performace.py:167]: comment

File: deepchecks/nlp/checks/model_evaluation/train_test_performance.py (hunk 2 above)
Comment (nirhutnik) [reply deepchecks/nlp/checks/model_evaluation/train_test_performance.py:36]: It's about significance as well. you'll have a drop of 10% for a change in one sample out of 10. I think 10 is too low, and 30 is a common min sample size.

Comment (nirhutnik) [reply deepchecks/nlp/checks/model_evaluation/train_test_performance.py:36]: I mean, in general, if my metrics reduced on such a small class, I also wouldn't care...

File: deepchecks/nlp/checks/model_evaluation/train_test_performance.py (hunk 1 above)
Comment (nirhutnik) [reply deepchecks/nlp/checks/model_evaluation/train_test_performance.py:42]: Also, the issue is - by which metric? And for each graph separately? It really makes it complicatd :/
